src/components/generic/TreeView.tsx

```diff
@@ -84,85 +83,103 @@ class TreeView extends React.Component<Props, State> {
-      api.getWpV2Pages({}),
+    const [pages, page, post, schools] = await Promise.all([
+      api.getWpV2Pages({per_page: 100}),
       api.getWpV2Pages({ lang: [ lang ], slug: [ slug ] }),
       api.getWpV2Posts({ lang: [ lang ], slug: [ slug ] }),
-      api.getMenusV1LocationsById({ lang: this.props.lang, id: "main" }),
       api.getWpV2CustomTaxonomy({ name: "schools" })
     ]);
     this.setState({
       pages: pages,
       page: page.length > 0 ? page[0] : undefined,
       post: post.length > 0 ? post[0] : undefined,
-      school: schools.find(item => `${ item.id }` === (page.length > 0 && page[0].taxonomy_schools && page[0].taxonomy_schools.length > 0 ? page[0].taxonomy_schools.join("") : "")),
-      currentPageOrPostId: String((page) ? page[0].id : ((post) ? post[0].id : undefined))
+      school: schools.find((item: any) => `${ item.id }` === (page.length > 0 && page[0].taxonomy_schools && page[0].taxonomy_schools.length > 0 ? page[0].taxonomy_schools.join("") : "")),
     });
-    if (menus.items) {
-      menus.items.forEach((item) => {
-        this.formLinkTreeStructure(item.child_items || [], item, []);
-      });
-    }
+    this.buildTree(pages);
   }
 
   /**
-   * Finds current parent link and forms link tree structure from it
-   * 
-   * @param menuStructure menu structure
-   * @param original original menu structure
-   * @param opened keeps track of the opened nodes
+   * Builds link tree structure
+   *
+   * @param pages page array
    */
-  private formLinkTreeStructure = (menuStructure: MenuItemData[], original: MenuItemData, opened: string[]) => {
-    const { currentPageOrPostId, pages, school } = this.state;
-    let isSchoolPage: boolean;
-    menuStructure.forEach((menu) => {
-      if (pages && school) {
-        const page = pages.find((item) => `${ item.id }` === menu.object_id);
-        if (page) {
-          const parent = pages.find((item) => item.id === page.parent);
-          if (parent && parent.taxonomy_schools && parent.taxonomy_schools.length === 0 && page.taxonomy_schools && page.taxonomy_schools.length > 0 && page.taxonomy_schools[0] === school.id) {
-            isSchoolPage = true;
-            if (menu.child_items) {
-              this.formLinkTreeStructure(menu.child_items, menu, []);
-            } else {
-              this.setState({
-                title: menu.title,
-                treeData: [],
-                initialOpenNodes: []
-              });
-            }
-          }
+  private buildTree = (pages: Page[]) => {
+    const { page, school } = this.state;
+    const linkTreeStructure: LinkTreeStructure[] = [];
+    let mainPages = pages.filter((item) => item.parent === 0);
+    if (school) {
+      let mainPage = pages.find((item) => item.taxonomy_schools && item.taxonomy_schools.find((x) => x === school.id ));
+      if (mainPage) {
+        let parent = pages.find((item) => item.id === mainPage!.parent);
```

Comment:
No need for parentheses in single parameter function expressions
```suggestion
    let mainPages = pages.filter(item => item.parent === 0);
    if (school) {
      let mainPage = pages.find(item => item.taxonomy_schools && item.taxonomy_schools.find((x) => x === school.id ));
      if (mainPage) {
        let parent = pages.find(item => item.id === mainPage!.parent);
```